dotnet/src/SemanticKernel.Abstractions/Connectors/WebApi/Rest/RestApiOperationRunner.cs

```diff
@@ -84,7 +91,10 @@ public RestApiOperationRunner(HttpClient httpClient, AuthenticateRequestAsyncCal
             requestMessage.Content = payload;
         }
 
-        requestMessage.Headers.Add("User-Agent", "Microsoft-Semantic-Kernel");
+        if (this._userAgent != null)
```

Comment:
For strings, let's check "string.IsNullOrWhitespace" unless you want to support empty strings here.